lib/ansible/executor/task_executor.py

```diff
@@ -399,6 +396,34 @@ def _run_loop(self, items):
 
         return results
 
+    def _calculate_delegate_to(self, templar, variables):
+        if self._task.delegate_to:
```

Comment:
shouldnt we just get 'validated value' here?

Reply:
Not at this point, `self._task.post_validate()` happens about 100 lines after the call to `_self._calculate_delegate_to`.  So part of the purpose of this function is to effectively change the `self._delegate_to` to a pre-calculated version before `post_validate()` runs when done in a loop.